pkg/clusteragent/admission/mutate/autoinstrumentation/imageresolver/config.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+// Package imageresolver provides configuration and utilities for resolving
+// container image references from mutable tags to digests.
+package imageresolver
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+)
+
+// RemoteConfigClient defines the interface we need for remote config operations
+type RemoteConfigClient interface {
+	GetConfigs(product string) map[string]state.RawConfig
+	Subscribe(product string, callback func(map[string]state.RawConfig, func(string, state.ApplyStatus)))
+}
+
+// Config contains information needed to create an ImageResolver
+type Config struct {
+	Site           string
+	DDRegistries   map[string]struct{}
+	RCClient       RemoteConfigClient
+	MaxInitRetries int
+	InitRetryDelay time.Duration
+}
+
+// NewConfig creates a new Config
+func NewConfig(cfg config.Component, rcClient RemoteConfigClient) Config {
```

Comment:
This could use a simple TDD test to assert the passed in config produces the expected struct